frame/elections-phragmen/src/migrations/unlock_and_unreserve_all_funds.rs

```diff
@@ -0,0 +1,433 @@
+
+impl<T: crate::Config> UnlockAndUnreserveAllFunds<T>
+where
+	BalanceOf<T>: Sum,
+{
+	/// Calculates and returns the total amounts deposited and staked by each account in the context
+	/// of this pallet.
+	///
+	/// The deposited and staked amounts are returned in two separate `BTreeMap` collections.
+	///
+	/// The first `BTreeMap`, `account_deposited_sums`, contains each account's total amount
+	/// deposited. This includes deposits made by Members, RunnerUps, Candidates, and Voters.
+	///
+	/// The second `BTreeMap`, `account_staked_sums`, contains each account's total amount staked.
+	/// This includes stakes made by Voters.
+	///
+	/// # Returns
+	///
+	/// This function returns a tuple of two `BTreeMap` collections and the weight of the reads:
+	///
+	/// * `BTreeMap<T::AccountId, BalanceOf<T>>`: Map of account IDs to their respective total
+	///   deposit sums.
+	/// * `BTreeMap<T::AccountId, BalanceOf<T>>`: Map of account IDs to their respective total
+	///   staked sums.
+	/// * `frame_support::weights::Weight`: The weight of reading the storage.
+	fn get_account_deposited_and_staked_sums() -> (
+		BTreeMap<T::AccountId, BalanceOf<T>>,
+		BTreeMap<T::AccountId, BalanceOf<T>>,
+		frame_support::weights::Weight,
+	) {
+		use crate::Voting;
+		use sp_runtime::Saturating;
+
+		let members = crate::Members::<T>::get();
+		let runner_ups = crate::RunnersUp::<T>::get();
+		let candidates = crate::Candidates::<T>::get();
+
+		// Get the total amount deposited (Members, RunnerUps, Candidates and Voters all can have
+		// deposits).
+		let account_deposited_sums: BTreeMap<T::AccountId, BalanceOf<T>> = members
+			// Massage all data structures into (account_id, deposit) tuples.
+			.iter()
+			.chain(runner_ups.iter())
+			.map(|member| (member.who.clone(), member.deposit))
+			.chain(candidates.iter().map(|(candidate, amount)| (candidate.clone(), *amount)))
+			.chain(
+				Voting::<T>::iter().map(|(account_id, voter)| (account_id.clone(), voter.deposit)),
+			)
+			// Finally, aggregate the tuples into a Map.
+			.fold(BTreeMap::new(), |mut acc, (id, deposit)| {
+				acc.entry(id.clone()).or_insert(Zero::zero()).saturating_accrue(deposit);
+				acc
+			});
+
+		// Get the total amount staked (only Voters stake) and count the number of voters.
+		let mut voters_len = 0;
+		let account_staked_sums: BTreeMap<T::AccountId, BalanceOf<T>> = Voting::<T>::iter()
+			.map(|(account_id, voter)| (account_id.clone(), voter.stake))
+			.fold(BTreeMap::new(), |mut acc, (id, stake)| {
+				voters_len.saturating_accrue(1);
+				acc.entry(id.clone()).or_insert(Zero::zero()).saturating_accrue(stake);
+				acc
+			});
+
+		(
+			account_deposited_sums,
+			account_staked_sums,
+			T::DbWeight::get().reads(
+				members
+					.len()
+					.saturating_add(runner_ups.len())
+					.saturating_add(candidates.len())
+					.saturating_add(voters_len.saturating_mul(T::MaxVotesPerVoter::get() as usize))
```

Comment:
Not sure why you multiply this?

Reply:
Each `Voter` contains a `Vec` called `votes` up to size `MaxVotesPerVoter`.

I guessed that the weight of the read scales with the size of the Vec, but maybe that's incorrect if it can all be loaded into memory in one read regardless?